projects/js-packages/charts/src/components/bar-chart/stories/index.docs.mdx

```diff
@@ -306,6 +306,51 @@ Display legends vertically for charts with many series:
 	/>` }
 />
 
+### Composition API
+
+For more flexible legend positioning and future extensibility, use the composition API with `<BarChart.Legend />` as a child component:
+
+<Canvas of={ BarChartStories.WithCompositionLegend } />
+
+<Source
+	language="jsx"
+	code={ `<BarChart
+		data={data}
+		withTooltips={true}
+		gridVisibility="x"
+	>
+		<BarChart.Legend
+			orientation="horizontal"
+			alignmentHorizontal="center"
+			alignmentVertical="bottom"
+		/>
+	</BarChart>` }
+/>
+
+**Benefits of the composition API:**
+- **Flexible positioning**: Place legends anywhere within the chart container
+- **Future extensibility**: Enables additional chart components like annotations
+- **Consistent API**: Same pattern as LineChart for a unified developer experience
+- **Backward compatibility**: Existing `showLegend` prop continues to work
+
+**Usage patterns:**
+
+<Source
+	language="jsx"
+	code={ `// Composition API (recommended for new code)
+	<BarChart data={data}>
+		<BarChart.Legend orientation="vertical" alignmentHorizontal="right" />
+	</BarChart>
+
+	// Traditional prop-based approach (still supported)
+	<BarChart
+		data={data}
+		showLegend={true}
+		legendOrientation="vertical"
```

Comment:
needs alignment after https://github.com/Automattic/jetpack/pull/44747

Reply:
yup, I'm on it. Will ship after resolving conflicts then create a new issue to iterate and consider the issues mentioned.